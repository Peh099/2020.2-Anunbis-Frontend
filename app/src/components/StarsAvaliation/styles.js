import styled from 'styled-components';
import Star_Icon_C from '../../assets/images/Star_Icon_C.svg'
import Star_Icon_BW from '../../assets/images/Star_Icon_BW.svg'

export const Container = styled.div`
  display: flex;

    span{
        background-color: var(--transparent);
        background-image: url(${Star_Icon_BW});
    }

    label:hover input ~ span {
    background-color: var(--transparent);
    background-image: url(${Star_Icon_C});
    }

    input:checked ~ span {
        background-color: var(--transparent);
        background-image: url(${Star_Icon_C});
    }

    input:checked ~ span:after {
      content: none;
    }
`;


