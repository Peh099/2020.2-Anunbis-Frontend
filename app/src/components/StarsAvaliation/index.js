import React from 'react';

import { Container } from './styles';
import Checkbox from '../Checkbox/'

function StarsAvaliation({ name, register }) {
    const [isChecked1, setChecked1] = React.useState(false);
    const [isChecked2, setChecked2] = React.useState(false);
    const [isChecked3, setChecked3] = React.useState(false);
    const [isChecked4, setChecked4] = React.useState(false);
    const [isChecked5, setChecked5] = React.useState(false);

    function setChecked(num, isChecked) {
        switch(num) {
            case 1:
                setChecked1(isChecked)
                break;
            case 2:
                setChecked2(isChecked)
                break;
            case 3:
                setChecked3(isChecked)
                break;
            case 4:
                setChecked4(isChecked)
                break;
            case 5:
                setChecked5(isChecked)
                break;

            default:
        }
    }
    function setStatus(num, status) {
        if(status === false){
            for(var a = num+1; a <= 5; a++) setChecked(a, status);
        }else{
            for(var b = 1; b <= num; b++) setChecked(b, status);
        }
    }

  return (
    <Container>
        <Checkbox name={name} register={register} checked={isChecked1} onChange={() => setStatus(1 ,!isChecked1)}/>
        <Checkbox name={name} register={register} checked={isChecked2} onChange={() => setStatus(2 ,!isChecked2)}/>
        <Checkbox name={name} register={register} checked={isChecked3} onChange={() => setStatus(3 ,!isChecked3)}/>
        <Checkbox name={name} register={register} checked={isChecked4} onChange={() => setStatus(4 ,!isChecked4)}/>
        <Checkbox name={name} register={register} checked={isChecked5} onChange={() => setStatus(5 ,!isChecked5)}/>
    </Container>
  );
}

export default StarsAvaliation;