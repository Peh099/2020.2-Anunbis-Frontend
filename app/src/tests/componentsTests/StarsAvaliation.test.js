import renderer from 'react-test-renderer';
import StarsAvaliation from '../../components/StarsAvaliation';

describe('Snapshot StarsAvaliation component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<StarsAvaliation />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})