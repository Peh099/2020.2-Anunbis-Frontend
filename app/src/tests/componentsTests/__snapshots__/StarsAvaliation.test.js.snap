// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot StarsAvaliation component matches the snapshot 1`] = `
<div
  className="sc-bdnxRM eAqkRU"
>
  <label
    className="sc-dlnjwi itVsq"
  >
    <input
      checked={false}
      data-testid="input-stars-1"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-gtsrHT ewUAES"
    />
  </label>
  <label
    className="sc-dlnjwi itVsq"
  >
    <input
      checked={false}
      data-testid="input-stars-2"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-gtsrHT ewUAES"
    />
  </label>
  <label
    className="sc-dlnjwi itVsq"
  >
    <input
      checked={false}
      data-testid="input-stars-3"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-gtsrHT ewUAES"
    />
  </label>
  <label
    className="sc-dlnjwi itVsq"
  >
    <input
      checked={false}
      data-testid="input-stars-4"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-gtsrHT ewUAES"
    />
  </label>
  <label
    className="sc-dlnjwi itVsq"
  >
    <input
      checked={false}
      data-testid="input-stars-5"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="sc-gtsrHT ewUAES"
    />
  </label>
</div>
`;
